devito/passes/iet/dtypes.py

```diff
@@ -1,14 +1,54 @@
 import numpy as np
 import ctypes
 
-from devito.ir import FindSymbols, Uxreplace
+from devito.ir import FindSymbols
+from devito.ir.iet.nodes import Dereference
+from devito.ir.iet.visitors import Uxreplace
+from devito.symbolics.extended_dtypes import Float16P
+from devito.tools.utils import as_list
+from devito.types.basic import Symbol
 
-__all__ = ['lower_complex']
+__all__ = ['lower_dtypes']
 
 
-def lower_complex(iet, lang, compiler):
+def lower_dtypes(iet, lang, compiler, sregistry):
     """
-    Add headers for complex arithmetic
+    Lowers float16 scalar types to pointers since we can't directly pass their
+    value. Also includes headers for complex arithmetic if needed.
+    """
+
+    iet, metadata = _complex_includes(iet, lang, compiler)
+
+    # Lower float16 parameters to pointers and dereference
+    body_prefix = []
+    body_mapper = {}
```

Comment:
I think we do because the symbol in the parameters needs to be mapped to one with a different dtype but same name; in the body the same symbol has to be mapped to a new one with a different name (which we assign by the `Dereference`)